script.service.singealarmclock/plugin.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+#     Copyright (C) 2012 Team-XBMC
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#    This script is based on service.skin.widgets
+#    Thanks to the original authors
+
+import xbmc, xbmcaddon
+import os, sys
+__cwd__      = xbmc.translatePath( xbmcaddon.Addon().getAddonInfo('path') ).decode("utf-8")
+__resource__ = xbmc.translatePath( os.path.join( __cwd__, 'resources', 'lib' ) ).decode("utf-8")
+sys.path.append(__resource__)
+
+import sys
+import xbmcgui
+import xbmcplugin
+import logging
+import urlparse
+
+import settings
+
+ADDON = xbmcaddon.Addon()
+ADDON_VERSION = ADDON.getAddonInfo('version')
+ADDON_NAME = ADDON.getAddonInfo('name')
+ADDON_LANGUAGE = ADDON.getLocalizedString
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+logging.basicConfig(format='[%(levelname)-7s] %(asctime)s %(name)s: %(message)s')
+
+
+def join_qs(**query_values):
+    pairs = ['%s=%s' % kv for kv in list(query_values.items())]
+    return '?' + '&'.join(pairs)
+
+
+def menu_item(plugin, label, **query_values):
+    list_item = xbmcgui.ListItem(label)
+    list_item.setLabel(label)
+    list_item.setLabel2(label)
+    list_item.setProperty('IsPlayable', 'false')
+    list_item.setThumbnailImage(__cwd__ + '/icon.png')
```

Comment:
drop `setThumbnailImage` and `setIconImage`, they'll result in NOP now in matrix